src/scripts/main.js

```diff
@@ -1 +1,67 @@
 'use strict';
+
+new Promise((resolve, reject) => {
+  const timeoutId = setTimeout(() => {
+    // eslint-disable-next-line prefer-promise-reject-errors
+    reject('First promise was rejected');
+  }, 3000);
+
+  document.body.addEventListener(
+    'click',
+    (e) => {
+      resolve('First promise was resolved');
+      clearTimeout(timeoutId);
+    },
+    { once: true },
+  );
+})
+  .then((successResults) => {
+    createNotification(successResults, true);
+  })
+  .catch((errorResult) => {
+    createNotification(errorResult, false);
+  });
+
+new Promise((resolve) => {
+  document.addEventListener('click', () => {
+    resolve('Second promise was resolved');
+  });
+
+  document.addEventListener('contextmenu', () => {
+    resolve('Second promise was resolved');
+  });
+}).then((successResults) => {
+  createNotification(successResults, true);
+});
```

Comment:
The second Promise is resolved on either 'click' or 'contextmenu' event. However, once one of these events is fired, the other event listener is still active and can lead to multiple resolutions of the same Promise. It's better to remove event listeners once the Promise has been resolved.